core/modules/wmgr/python/procMgr.py

```
@@ -0,0 +1,146 @@
+        subprocess.check_call(cmd, close_fds=True)
+        return 0
+    except subprocess.CalledProcessError as exc:
+        _log.debug('subprocess exception: %s', exc)
+        if noexcept:
+            return exc.returncode
+        else:
+            # re-throw
+            raise
+
+#------------------------
+# Exported definitions --
+#------------------------
+
+procService = Blueprint('procService', __name__, template_folder='procService')
+
+
+@procService.route('', methods=['GET'])
+def services():
+    """ Return the list of services """
+
+    _log.debug('request: %s', request)
+
+    services = [_svcDict(svc) for svc in _svcNames]
+
+    return json.jsonify(results=services)
+
+
+@procService.route('/<service>', methods=['GET'])
+def state(service):
+    """
+    Return service state.
+    """
+
+    _log.debug('request: %s', request)
+
+    if service not in _svcNames:
+        raise ExceptionResponse(404, "InvalidService", "Invalid service name " + service)
+
+    runDir = Config.instance().runDir
+    initScript = os.path.join(runDir, 'etc/init.d', service)
+
+    cmd = [initScript, 'status']
+    running = _runCmd(cmd) == 0
+
+    return json.jsonify(results=_svcDict(service, running))
+
+
+@procService.route('/<service>', methods=['PUT'])
+def action(service):
+    """
+    Do something with service.
+
+    Following parameters are expected to come in a request (in request body
+    with application/x-www-form-urlencoded content like regular form):
+        action: one of 'stop', 'start', 'restart' (required)
+    """
+
+    _log.debug('request: %s', request)
+
+    if service not in _svcNames:
+        raise ExceptionResponse(404, "InvalidService", "Invalid service name " + service)
+
+    action = request.form.get('action', '').strip()
+    if not action:
+        raise ExceptionResponse(400, "MissingArgument", "Action argument (action) is missing")
+    if action not in ['start', 'stop', 'restart']:
```

My mistake, the line below should raise different exception (InvalidArgument instead of MissingArgument). I'll fix it.